pkg/fields/selector.go

```diff
@@ -219,29 +236,43 @@ func try(selectorPiece, op string) (lhs, rhs string, ok bool) {
 func parseSelector(selector string, fn TransformFunc) (Selector, error) {
 	parts := strings.Split(selector, ",")
 	sort.StringSlice(parts).Sort()
-	var items []Selector
+
+	result := NewSelector()
 	for _, part := range parts {
 		if part == "" {
 			continue
 		}
+		var (
+			requirement *selectors.Requirement
+			err         error
+		)
 		if lhs, rhs, ok := try(part, "!="); ok {
-			items = append(items, &notHasTerm{field: lhs, value: rhs})
+			requirement, err = NewRequirement(lhs, selectors.NotEqualsOperator, sets.NewString(rhs))
 		} else if lhs, rhs, ok := try(part, "=="); ok {
-			items = append(items, &hasTerm{field: lhs, value: rhs})
+			requirement, err = NewRequirement(lhs, selectors.DoubleEqualsOperator, sets.NewString(rhs))
 		} else if lhs, rhs, ok := try(part, "="); ok {
-			items = append(items, &hasTerm{field: lhs, value: rhs})
+			requirement, err = NewRequirement(lhs, selectors.EqualsOperator, sets.NewString(rhs))
 		} else {
 			return nil, fmt.Errorf("invalid selector: '%s'; can't understand '%s'", selector, part)
 		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		result = result.Add(*requirement)
 	}
-	if len(items) == 1 {
-		return items[0].Transform(fn)
-	}
-	return andTerm(items).Transform(fn)
+	return result.Transform(fn)
 }
 
 // OneTermEqualSelector returns an object that matches objects where one field/field equals one value.
 // Cannot return an error.
 func OneTermEqualSelector(k, v string) Selector {
-	return &hasTerm{field: k, value: v}
+	result := NewSelector()
+	requirement, err := NewRequirement(k, selectors.EqualsOperator, sets.NewString(v))
+	if err != nil {
+		panic(err)
```

Comment:
Not sure about this `panic`,  the previous code couldn't produce errors but the set based requirement is different. I would prefer to return an error modifying invocation code.